src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -223,6 +223,10 @@ private static bool IsTupleExpression(BoundKind kind)
             {
                 var field = fields[i];
 
+                // Use default field rather than implicitly named fields since
+                // fields from inferred names are not usable in C# 7.0.
+                field = field.CorrespondingTupleField ?? field;
```

Comment:
Thanks @vsadov, an inaccessible tuple field hits this code path. I've added a use-site test.